### PR TITLE
Added a promise to writeImageToFile

### PR DESCRIPTION
**Pull request by hparadiz**
Added a promise to writeImageToFile which resolves when the finish event has fired from the write stream thereby fixing a race condition.

### CODE REVIEW

**Comment by hparadiz** [comment]
So the issue is with the style
```
One space required after "function" keyword at index.js :
   148 |            return new BbPromise((resolve,reject) => {
   149 |                var out = fs.createWriteStream(filePath);
   150 |                out.on('finish', function() {
-------------------------------------------------^
   151 |                    return resolve(this);
   152 |                });
One space required after "function" keyword at index.js :
   151 |                    return resolve(this);
   152 |                });
   153 |                out.on('error', function(err) {
------------------------------------------------^
   154 |                    return reject(err);
   155 |                });
>> 2 code style errors found!
```

Thing is I'm not sure how to run travis-ci before running the pull request. Is that the only thing keeping this from being approved?

**Comment by hparadiz** [comment]
Any word on this? Did you want me to submit another PR with the code style fixes?

**Comment by sedouard** [comment]
@hparadiz  sorry for the delay running `grunt jscs` should do the trick. Or just update the code to add the spaces

File: index.js [hunk 1]
```diff
@@ -145,8 +145,17 @@ class ChartjsNode {
     writeImageToFile(imageType, filePath) {
         return this.getImageBuffer(imageType)
         .then(buffer => {
-            var out = fs.createWriteStream(filePath);
-            return out.write(buffer);
+            return new BbPromise((resolve,reject) => {
+                var out = fs.createWriteStream(filePath);
+                out.on('finish', function() {
```
Comment (hparadiz) [inline index.js]: I dunno why I did that to be honest. Agreed -> resolve() should be just fine

**Comment by hparadiz** [comment]
Should be good to go now! 👍

**Comment by sedouard** [comment]
@hparadiz can you take a peek at the review commends? Just a couple tweaks

**Comment by sedouard** [comment]
@hparadiz Thanks!

**Pull request merged**